cpp/include/raft/handle.hpp

```diff
@@ -52,56 +53,47 @@ class handle_t {
   static constexpr int kNumDefaultWorkerStreams = 0;
 
  public:
+  // delete copy/move constructors and assignment operators as
+  // copying and moving underlying resources is unsafe
+  handle_t(const handle_t&) = delete;
+  handle_t& operator=(const handle_t&) = delete;
+  handle_t(handle_t&&) = delete;
+  handle_t& operator=(handle_t&&) = delete;
+
   /**
-   * @brief Construct a handle with the specified number of worker streams
+   * @brief Construct a handle with a stream view and stream pool
    *
-   * @param[in] n_streams number worker streams to be created
+   * @param[in] stream the default stream (which has the default value of nullptr if unspecified)
```

Comment:
The use of the term `default_stream` might be confusing me a bit. We had originally been using the CUDA default (e.g. device synchronous) stream everywhere and had to go through quite a clean up job to replace it with an explicit stream everywhere. Is this the CUDA device-synchronous default stream which is set in the kernels when stream == nullptr or does the stream view somehow make that into an explicit stream? Or has our ruling changed on the topic of using the device synchronous default stream within raft/cuml? 

If it is the case that we plan to allow the device synchronous default stream, we probably should update the cuml docs. Also it's worth making sure that we continue to account for a stream in the code everywhere applicable. Tagging @teju85 just to make sure there weren't any other reasons for that change initially.

Reply:
At some point, this came back I believe because `raft::handle_t` was creating a handle with `stream == nullptr` even before updates by this PR